integration/integration.go

```diff
@@ -176,6 +176,21 @@ func (i *Integration) Logger() log.Logger {
 	return i.logger
 }
 
+// FindEntity finds ad return an entity by name. returns false if entity does not exist in the integration
+func (i *Integration) FindEntity(name string) (*Entity, bool) {
+	for _, e := range i.Entities {
+		// nil metadata usually means the "host" entity, so keep searching
+		if e.Metadata == nil {
```

Comment:
Entities is a slice of pointers... can `e` itself being nil or it is safe to take the assumption it will never happen?

Entities is exported so theoretically anyone can add an element to it

Reply:
I think we can assume entities is not nil. When we create a new integration it sets entities to an empty slice. 
Someone can always set the slice to nil and although  don't think that's a realistic scenario might as well check for that

Reply:
no I mean `e` a generic element of the slice, not the slice itself.

Reply:
Ah ok, i see what you mean.. It's a possibility, but then someone is doing the wrong thing almost on purpose..

Reply:
> no I mean `e` a generic element of the slice, not the slice itself. Entities is defined as `Entities []*Entity` so elements can generate nullpointerExceptions

That's why it's a bad idea to have these things exported like this, it let's someone do the wrong things